src/components/ChatContent/Messages/BlockMenu/ReportDialog.vue

```diff
@@ -0,0 +1,300 @@
+      <div style="display: inline-block; margin: 0px 12px;">
+        <input type="checkbox" v-model="report">
+        <label for="report-report">Report User</label>
+      </div>
+    </div>
+    <div style="margin-top: 15px;" v-show="report">
+      <label for="report-message">Additional Comments:</label>
+      <textarea v-model="reportComments" style="resize: none; height: 60px;"></textarea>
+    </div>
+    <div style="margin-top: 15px;">
+      <div
+        class="blue-button blue-button-interactive"
+        style="width: 40%; display: inline-block; vertical-align: middle;"
+        @click="closeModal"
+      >Cancel</div>
+      <div
+        class="green-button green-button-interactive"
+        style="width: 40%; display: inline-block; float: right; vertical-align: middle;"
+        @click="submit"
+      >Continue</div>
+    </div>
+  </modal>
+</template>
+
+<script lang="ts">
+  import { Component, Prop } from 'vue-property-decorator';
+  import Vue from '@/types/vue';
+  import { Client } from 'irc-framework';
+  import parseUsername from '@/tools/parseUsername';
+  import User from '../../../../types/user';
+  import Message from '../../../../types/message';
+
+@Component({})
+  export default class ReportDialog extends Vue {
+  report = false;
+
+  ignore = false;
+
+  message: Message | null = null;
+
+  get userToReport(): User {
+    return this.message ? this.message.user : User.annonymous;
+  }
+
+  $refs!: {
+    reportModal: HTMLFormElement;
+  };
+
+  reportComments: string = '';
+
+  submit() {
+    if (this.ignore) {
+      this.$store.commit('ignore', this.userToReport.username);
+    }
+    const client = this.$store.state.client!;
+    if (this.report) {
+      client.say(
+        '#ops-notifications',
+        `[REPORT] Reporting ${this.userToReport.username} (${
+          this.userToReport.uid
+        }) by ${this.$store.state.currentUser.username} (${
+          this.$store.state.currentUser.uid
+        }).\r\n`,
+      );
+      if (this.message) {
+        client.say(
+          '#ops-notifications',
+          `[REPORTED MESSAGE] ${this.message!.message}\r\n`,
+        );
+      }
+      client.say(
+        '#ops-notifications',
+        `[REPORT REASON] ${this.reportComments}\r\n`,
+      );
+    }
+    this.closeModal();
+  }
+
+  closeModal() {
+    this.$modal.hide('reportModal');
+    this.reportComments = '';
+  }
+
+  created() {
+    this.$store.subscribe((mutation, state) => {
+      if (mutation.type === 'openReportModal') {
+        this.message = mutation.payload.message;
+        this.report = mutation.payload.defaults.report;
+        this.ignore = mutation.payload.defaults.ignore;
+        this.$modal.show('reportModal');
+      }
+    });
+  }
+  }
+</script>
+
+<style lang="scss">
+/* Report dialog */
+.report-dialog {
+  background: #123768;
+  border-radius: 5px;
+  color: white;
+  padding: 20px;
+}
+</style>
+
+<style scoped lang="scss">
+a {
+  color: white;
+}
+
+.blue-button,
```

Comment:
I'll need to look more into that, but I believe I found it right to change some of the stuff here on the connection button, in order to make it so its position is where its shadow is located, and not where the top of the button is.
While I agree that making this css global is probably the best option, I don't really like the disconnect button being an edge case. We'll probably philosophize about it as part of the site rewrite though.

Reply:
Certainly - if there's something obvious that could be done to clean it up , do it, otherwise don't worry about it. I think fee free to merge things together into less classes and break with the CSS structure from the website, it's going to be changing significantly anyhow.